sqlmesh/core/model/meta.py

```diff
@@ -261,6 +265,19 @@ def _pre_root_validator(cls, values: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
                     f"Cannot use argument 'grain' ({grain}) with 'grains' ({grains}), use only grains"
                 )
             values["grains"] = ensure_list(grain)
+
+        table_properties = values.pop("table_properties", None)
+        if table_properties:
+            model_name = values.get("name")
```

Comment:
Nit: I think this is expected to always exist right? Given that `name` is required.
```suggestion
            model_name = values["name"]
```

Reply:
Yes, it feels like it should always exists as it's explicitly added in the provided validation fields. I'll add that change so that it fails eagerly if that behavior changes (thus preventing a silent switch to a `Python model "None"'` log).